src/main/java/org/elasticsearch/index/query/IdsQueryBuilder.java

```diff
@@ -181,24 +175,25 @@ public QueryValidationException validate() {
     }
 
     @Override
-    public void readFrom(StreamInput in) throws IOException {
-        this.types = in.readStringList();
-        this.ids = in.readStringList();
-        queryName = in.readOptionalString();
-        boost = in.readFloat();
+    public IdsQueryBuilder readFrom(StreamInput in) throws IOException {
+        IdsQueryBuilder idsQueryBuilder = new IdsQueryBuilder(in.readStringArray());
+        idsQueryBuilder.addIds(in.readStringArray());
+        idsQueryBuilder.queryName = in.readOptionalString();
+        idsQueryBuilder.boost = in.readFloat();
+        return idsQueryBuilder;
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
-        out.writeStringList(this.types);
-        out.writeStringList(this.ids);
```

Comment:
I think at this point I will also remove the readStringList and writeStringList that we previously added to master if you don't mind @cbuescher :)